agent/engine/task_manager.go

```diff
@@ -407,6 +407,12 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 	if event.Status <= containerKnownStatus {
 		seelog.Infof("Managed task [%s]: redundant container state change. %s to %s, but already %s",
 			mtask.Arn, container.Name, event.Status.String(), containerKnownStatus.String())
+
+		// Only update container metadata when status keeps the same, ignore updating metadata if there
+		// is backward transition
+		if event.Status == containerKnownStatus {
```

Comment:
should we explicitly check for RUNNING status if the use case covers only that?

Reply:
I will update it to start from only covering RUNNING since so far I haven't found out other use cases.